core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -61,19 +86,51 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
+      throw e.toThrift();
+    } catch (IOException e) {
+      LOG.debug("I/O Error: {}", callable, e);
+      throw new ThriftIOException(e.getMessage());
+    } catch (Exception e) {
+      LOG.error("Unexpected Error: {}", callable, e);
+      throw new UnexpectedAlluxioException(e).toThrift();
+    }
+  }
+
+  /**
+   * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown. This
+   * method also logs enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   * @throws ThriftIOException if the callable throws an IOException
+   */
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
```

Comment:
If you guys think it is valuable to have enter/exit logging for periodic RPCs like `sessionHeartbeat`, we could remove the non-logging version all together. However I just didn't think it would be practical (ie. 100 clients connected to a worker means ~200 log lines per second).